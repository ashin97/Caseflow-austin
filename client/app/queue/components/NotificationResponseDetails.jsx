import React, { useState } from 'react';

const NotificationResponseDetails = () => {
  const [response, setResponse] = useState('-');
  const [date, setDate] = useState('-');
  const [time, setTime] = useState('-');

  const tableCols = ['Appellant Acknowledgement', 'Response Date', 'Response Time', '', '', ''];

  const renderTable = (cols) => {
    return cols.map((col) => <td><strong>{col}</strong></td>);
  };

  return (
    <tr >
      {renderTable(tableCols)}
    </tr>
  );
};

export default NotificationResponseDetails;
