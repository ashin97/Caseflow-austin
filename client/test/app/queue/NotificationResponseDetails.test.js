import React from 'react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import NotificationResponseDetails from '../../../app/queue/components/NotificationResponseDetails';

describe('NotificationResponseDetails', () => {
  const appellantAcknowledgement = 'Yes';
  const responseDate = '12/26/2023';
  const responseTime = '12:00';
  const appellantAcknowledgementDenial = 'No';

  const setup = (response, date, time) => {
    return render(<NotificationResponseDetails response={response} date={date} time={time} />);
  };

  it('renders "Yes" in Appellant Acknowledgement', () => {
    const { container } = setup(appellantAcknowledgement, responseDate, responseTime);
    const cell = container.querySelectorAll('td')[6];

    expect(cell.textContent).toBe('Yes');
  });

  it('renders "12/26/2023" in Response Date', () => {
    const { container } = setup(appellantAcknowledgement, responseDate, responseTime);
    const cell = container.querySelectorAll('td')[7];

    expect(cell.textContent).toBe('12/26/2023');
  });

  it('renders "12:00" in Response Time', () => {
    const { container } = setup(appellantAcknowledgement, responseDate, responseTime);
    const cell = container.querySelectorAll('td')[8];

    expect(cell.textContent).toBe('12:00');
  });

  it('renders "No" in Appellant Acknowledgement', () => {
    const { container } = setup(appellantAcknowledgementDenial, responseDate, responseTime);
    const cell = container.querySelectorAll('td')[6];

    expect(cell.textContent).toBe('No');
  });

  it('matches snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('Virtual passes ally testing', async () => {
    const { container } = setup();
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
