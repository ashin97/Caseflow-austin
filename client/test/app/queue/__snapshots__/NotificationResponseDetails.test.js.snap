// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationResponseDetails matches snapshot 1`] = `
<div>
  <tr
    style="border-bottom: hidden;"
  >
    <td>
      <strong>
        Appellant Acknowledgement
      </strong>
    </td>
    <td>
      <strong>
        Response Date
      </strong>
    </td>
    <td>
      <strong>
        Response Time
      </strong>
    </td>
    <td>
      <strong />
    </td>
    <td>
      <strong />
    </td>
    <td>
      <strong />
    </td>
  </tr>
  <tr>
    <td />
    <td />
    <td />
    <td />
    <td />
    <td />
  </tr>
</div>
`;
